src/app/components/user/avatar-modal/UserAvatarModal.vue

```diff
@@ -0,0 +1,49 @@
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue';
+import { run } from '../../../../utils/utils';
+import { Api } from '../../../../_common/api/api.service';
+import AppButton from '../../../../_common/button/AppButton.vue';
+import AppModal from '../../../../_common/modal/AppModal.vue';
+import { useModal } from '../../../../_common/modal/modal.service';
+import { useCommonStore } from '../../../../_common/store/common-store';
+import FormUserAvatarFrame from '../../../../_common/user/user-avatar/frame/FormUserAvatarFrame.vue';
+import FormAvatar from '../../forms/avatar/avatar.vue';
+
+const modal = useModal()!;
+const { user } = useCommonStore();
+
+const hasAvatarFrameSelector = ref(false);
+
+onMounted(() => {
+	run(async () => {
```

Comment:
i think you can pass an async funciton into onMounted just fine without issues. no need for run()